js/main.js

```diff
@@ -82,37 +85,204 @@ var renderContent = function () {
 
 renderContent();
 
-function getCommentForBigPicture(arr) {
-  var commentBigPictureElement = document.querySelector('#big_picture').content;
+function renderCommentForBigPicture(commentBigPictureElement, comment) {
   var commentElement = commentBigPictureElement.cloneNode(true);
-  commentElement.querySelector('.social__picture').src = arr.avatar;
-  commentElement.querySelector('.social__picture').alt = arr.name;
-  commentElement.querySelector('.social__text').textContent = arr.message;
+  commentElement.querySelector('.social__picture').src = comment.avatar;
+  commentElement.querySelector('.social__picture').alt = comment.name;
+  commentElement.querySelector('.social__text').textContent = comment.message;
   return commentElement;
 }
 
+function closeBigPicture() {
+  var bigPictureElement = document.querySelector('.big-picture');
+  bigPictureElement.classList.add('hidden');
+  document.querySelector('body').classList.remove('modal-open');
+}
+
 function renderBigPicture(photo) {
   var bigPictureElement = document.querySelector('.big-picture');
   var bigPictureImgElement = document.querySelector('.big-picture__img img');
   var likesCountElement = document.querySelector('.likes-count');
   var socialCaptionElement = document.querySelector('.social__caption');
   var commentsCountElement = document.querySelector('.comments-count');
   var socialCommentsElement = document.querySelector('.social__comments');
-  var socialCommentCountElement = document.querySelector('.social__comment-count');
-  var commentsLoaderElement = document.querySelector('.comments-loader');
+  document.getElementById('picture-cancel').onclick = closeBigPicture;
 
   bigPictureElement.classList.remove('hidden');
   bigPictureImgElement.src = photo.url;
   likesCountElement.textContent = photo.likes;
   socialCaptionElement.textContent = photo.description;
   commentsCountElement.textContent = photo.comments.length;
-  for (var i = 0; i <= photo.comments.length; i++) {
-    socialCommentsElement.appendChild(getCommentForBigPicture(photo.comments[i]));
+  var template = document.querySelector('.social__comment').cloneNode(true);
+  document.querySelector('.social__comments').textContent = '';
+
+  for (var i = 0; i < photo.comments.length; i++) {
+    socialCommentsElement.appendChild(renderCommentForBigPicture(template, photo.comments[i]));
   }
```

Comment:
Тут должен быть итератор по массиву
```suggestion
photo.comments.forEach(function(comment){
    socialCommentsElement.appendChild(renderCommentForBigPicture(template, comment));
});
```